Utils/Dataflow/009_oracleConnector/dbConnection.py

```diff
@@ -121,10 +121,21 @@ def save_query_file(self, qname, src_filename, params=None):
             q = f.read().rstrip().rstrip(';')
             if isinstance(params, dict):
                 q = q % params
+            elif '%' in q.replace('%%', ''):
+                # Check for single '%' is done because a valid query without
+                # parameters (and their configuration) is possible.
+                sys.stderr.write("(WARN) No parameters were configured "
+                                 "for '%s' query. This can result in "
+                                 "'ORA-00911' error.\n" % qname)
             self.queries[qname]['query'] = q
         except IOError, err:
             sys.stderr.write("(ERROR) Failed to read query file: %s\n" % err)
             return False
+        except KeyError, err:
+            sys.stderr.write("(ERROR) Failed to update query: expected "
+                             "parameter %s not found in the "
+                             "configuration.\n" % err)
```

Comment:
When I read the log:
```
(INFO) Establish connection to the DB...
(INFO) Connection established.
(ERROR) Failed to update query: expected parameter 'production_or_analysis_cond' not found in the configuration.
(WARN) Failed to save query 'tasks'
(ERROR) Failed to update query: expected parameter 'production_or_analysis_cond' not found in the configuration.
(WARN) Failed to save query 'datasets'
(ERROR) Queries seem to be misconfigured. Exiting.
```
my first question was "what does it mean, ``update query''?"

Maybe make it less literary, like... 
```
(INFO) Establish connection to the DB...
(INFO) Connection established.
(ERROR) Query parameter missed: 'production_or_analysis_cond' (config section 'queries.params').
(WARN) Failed to save query 'tasks'
(ERROR) Query parameter missed: 'production_or_analysis_cond' (config section 'queries.params').
(WARN) Failed to save query 'datasets'
(ERROR) Queries seem to be misconfigured. Exiting.
```

Not that I think this variant is perfect, I wish it was better... so any rewording will be appreciated ;)
But the idea is to make error message less wordy (like `Query parameter missed`), while add more specific details (in which section of the config one should look for parameters). After all, `(ERROR)` itself supposes that some failure is encountered, so we can skip saying it explicitly.

Reply:
> my first question was "what does it mean, ``update query''?"

Oh, I see. Can indeed be confusing.

> add more specific details (in which section of the config one should look for parameters)

I thought about something like this, but then it occured to me that this is a method that applies a dictionary to a query, and knows nothing about config files or their sections. Would it be okay to mention them - if we consider the fact that it is possible in the future for this method to be called while taking parameters not from the config file?

How about this:

`(ERROR) Query parameter missing: 'production_or_analysis_cond'`

And we change the example config's description of 'queries.params' from

    # Parametric values for all queries:

to

    # Query parameters for all queries:

I think this should be enough to find the section in the config, and

`(ERROR) Queries seem to be misconfigured. Exiting.`

points to the config.

P.S. Other related messages should be changed to use the term `query parameter` too in such case.
P.P.S. Actually, the method knows nothing about Oracle errors as well, but it is preparing queries for Oracle and warns about a possible outcome - so this, I believe, is fine.

Reply:
> this is a method that applies a dictionary to a query, and knows nothing about config files or their sections. Would it be okay to mention them...?

Right you are, config file and sections belong to 009, while dbConnection seem to be a candidate for a move to (some) common library.

Then OK, let\`s leave only the first part, as you suggest, and update config example. One will possibly use his own config file without comments, yet there still is a proper config example; yet somewhere in configdocs (one day) it should be described as well; example is good, documentation is better.

P.S. The better solution, as I see now, would be to turn all errors and `return False` into exceptions. Then we could raise proper exception, something like `InvalidQueryParameter`, catch it in the main code and output proper log message, pointing to the section in config file. But it would require quite a refactoring, so... not this time, I believe?
Funny thing, when I wrote the code I intentionally tried to avoid exceptions... yet it is Pythonic style, indeed ;)

Reply:
Done.